File: apt/app/src/main/java/com/android/apt/MainActivity.java
package com.android.apt;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.android.annotation.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject("zhangsan")
    User user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: JetpackStudy/databinding/src/main/java/com/android/databinding/MainActivity.kt
package com.android.databinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.android.databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)

        val idol = Idol("Sceleet", 1)
        binding.idol = idol
        binding.eventhandle = EventHandleListener(this)
    }
}